book/src/development/define_lints.md

```diff
@@ -0,0 +1,177 @@
+# Define New Lints
+
+The journey of any new Clippy lint begins with one step.
+And this first step is to define the lint and register it to the codebase.
```

Comment:
I like this style of telling a story :) Though I would suggest collapsing these sentences:

```suggestion
The first step in the journey of a new lint is to define the definition and registration of the lint in Clippy's codebase.
```